R/convertOMLDataSetToMlr3.R

```diff
@@ -0,0 +1,123 @@
+#'   \dQuote{Survival Analysis}.
+#'   Default is \code{NULL} which means to guess it from the target column in the
+#'   data set. If that is a factor or a logical, we choose classification.
+#'   If it is numeric we choose regression. In all other cases an error is thrown.
+#' @param target [\code{character}]\cr
+#'   The target for the classification/regression task.
+#'   Default is the \code{default.target.attribute} of the \code{\link{OMLDataSetDescription}}.
+#' @param ignore.flagged.attributes [\code{logical(1)}]\cr
+#'   Should those features that are listed in the data set description slot \dQuote{ignore.attribute}
+#'   be removed?
+#'   Default is \code{TRUE}.
+#' @param drop.levels [\code{logical(1)}]\cr
+#'   Should empty factor levels be dropped in the data?
+#'   Default is \code{TRUE}.
+#' @param fix.colnames [\code{logical(1)}]\cr
+#'   Should colnames of the data be fixed using \code{\link[base]{make.names}}?
+#'   Default is \code{TRUE}.
+#' @template arg_verbosity
+#' @return [\code{\link[mlr3]{Task}}].
+#' @family data set-related functions
+#' @example /inst/examples/convertOMLDataSetToMlr3.R
+#' @export
+convertOMLDataSetToMlr3 = function(
+  obj,
+  mlr.task.id = "<oml.data.name>",
+  task.type = NULL,
+  target = obj$desc$default.target.attribute,
+  ignore.flagged.attributes = TRUE,
+  drop.levels = TRUE,
+  fix.colnames = TRUE,
+  verbosity = NULL) {
+
+  assertClass(obj, "OMLDataSet")
+  assertSubset(target, obj$colnames.new)
+  assertFlag(ignore.flagged.attributes)
+  assertFlag(drop.levels)
+
+  data = obj$data
+  desc = obj$desc
+
+  # no task type? guess it by looking at target
+  if (is.null(task.type))
+    task.type = guessTaskType(data[, target])
+  assertChoice(task.type, getValidTaskTypes())
+
+  #  remove ignored attributes from data
+  if (any(!is.na(desc$ignore.attribute)) & ignore.flagged.attributes) {
+    keep.cols = obj$colnames.old %nin% desc$ignore.attribute
+    data = data[, keep.cols, drop = FALSE]
+  }
+
+  # drop levels
+  if (drop.levels)
+    data = droplevels(data)
+
+  # fix colnames using make.names
+  if (fix.colnames) {
+    colnames(data) = make.names(colnames(data), unique = TRUE)
+    target = make.names(target, unique = TRUE)
+  }
+
+  # get fixup verbose setting for mlr
+  if (is.null(verbosity))
+    verbosity = getOMLConfig()$verbosity
+  fixup = ifelse(verbosity == 0L, "quiet", "warn")
+
+  mlr.task = switch(task.type,
+    "Supervised Classification" = mlr3::TaskClassif$new(id = desc$name, backend = data, target = target),
+    "Supervised Regression" = mlr3::TaskRegr$new(id = desc$name, backend = data, target = target),
+    "Survival Analysis" = mlr3survival::TaskSurv$new(id = desc$name, backend = data, target = target),
+   stopf("Encountered currently unsupported task type: %s", task.type)
+  )
+
+  if (!is.null(mlr.task.id))
+    mlr.task$id = replaceOMLDataSetString(mlr.task.id, obj)
+
+  return(mlr.task)
+}
+
+replaceOMLDataSetString = function(string, data.set) {
```

Comment:
this function already exists in `convertOMLDataSetToMlr.R`. unless we plan to remove mlr support this should not be duplicated. Same with `guessTaskType` and possibly others.